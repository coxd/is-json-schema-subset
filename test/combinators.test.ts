import './common';

describe('Schema combinators', () => {
  it('should accept oneOf', () =>
    expect({
      title: 'child',
      type: 'object',
      properties: {
        foo: { type: 'boolean' },
      },
      // @ts-ignore TS2339
    }).toSatisfy({
      title: 'parent',
      type: 'object',
      oneOf: [
        {
          type: 'object',
          properties: {
            foo: { type: 'boolean' },
          },
        },
        {
          type: 'object',
          properties: {
            foo: { type: 'string' },
          },
        },
      ],
    }));

  it('should reject oneOf: no match', () =>
    expect({
      title: 'child',
      type: 'object',
      properties: {
        foo: { type: 'number' },
      },
      // @ts-ignore TS2339
    }).toViolate({
      title: 'parent',
      type: 'object',
      oneOf: [
        {
          type: 'object',
          properties: {
            foo: { type: 'boolean' },
          },
        },
        {
          type: 'object',
          properties: {
            foo: { type: 'string' },
          },
        },
      ],
    }));

  it('should reject oneOf with multiple matches', () =>
    expect({
      title: 'child',
      type: 'object',
      // @ts-ignore TS2339
    }).toViolate({
      title: 'parent',
      type: 'object',
      oneOf: [
        {
          type: 'object',
          properties: {
            foo: { type: 'boolean' },
          },
        },
        {
          type: 'object',
          properties: {
            foo: { type: 'string' },
          },
        },
      ],
    }));

  it('should accept anyOf', () =>
    expect({
      title: 'child',
      type: 'object',
      properties: {
        foo: { type: 'boolean' },
      },
      // @ts-ignore TS2339
    }).toSatisfy({
      title: 'parent',
      type: 'object',
      anyOf: [
        {
          type: 'object',
          properties: {
            foo: { type: 'boolean' },
          },
        },
        {
          type: 'object',
          properties: {
            foo: { type: 'string' },
          },
        },
      ],
    }));

  it('should reject anyOf: no match', () =>
    expect({
      title: 'child',
      type: 'object',
      properties: {
        foo: { type: 'number' },
      },
      // @ts-ignore TS2339
    }).toViolate({
      title: 'parent',
      type: 'object',
      anyOf: [
        {
          type: 'object',
          properties: {
            foo: { type: 'boolean' },
          },
        },
        {
          type: 'object',
          properties: {
            foo: { type: 'string' },
          },
        },
      ],
    }));

  it('should accept anyOf: multiple matches', () =>
    expect({
      title: 'child',
      type: 'object',
      // @ts-ignore TS2339
    }).toSatisfy({
      title: 'parent',
      type: 'object',
      anyOf: [
        {
          type: 'object',
          properties: {
            foo: { type: 'boolean' },
          },
        },
        {
          type: 'object',
          properties: {
            foo: { type: 'string' },
          },
        },
      ],
    }));
});
