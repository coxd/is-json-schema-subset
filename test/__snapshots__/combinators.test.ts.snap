// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema combinators should reject oneOf: no match 1`] = `Array []`;
